util/include/rusefi/rusefi_time_types.h

```diff
@@ -24,7 +34,7 @@ using efitick_t = int64_t;
 /**
  * 64 bit time in microseconds (1/1_000_000 of a second), since boot
  */
-using efitimeus_t = int64_t;
+using efitimeus_t = tagged_numeric_t<timestamp_tag_s<int64_t, time_unit::MICROSECOND>>;
```

Comment:
can/should we just use std::chrono instead?

Reply:
Well, it makes sense. I will rewrite code using std::chrono::duration<int64_t, micros> and std::chrono::time_point.

Reply:
> Well, it makes sense. I will rewrite code using std::chrono::duration<int64_t, micros> and std::chrono::time_point.

What about our 1/168_000_000 second ticks and our 1/218_000_000 second tickets depending on target platform?

Reply:
> What about our 1/168_000_000 second ticks and our 1/218_000_000 second tickets depending on target platform?

```c++
// 4mhz was chosen because it's the GCD of (84, 108, 200), the three speeds of STM32 TIM5 clock currently supported
// https://www.wolframalpha.com/input/?i=common+factors+of+168+180+216
#define US_TO_NT_MULTIPLIER (4)
```

Reply:
ops https://github.com/rusefi/rusefi/commit/93edf728147714f666a29a2c57ed8ee19368d596